### PR TITLE
fix: simplify system migrator and verifier

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As part of https://github.com/ChainSafe/forest/issues/2801

Changes introduced in this pull request:

- Use macros to avoid repetitive code
- Define `XXState(Old|New)`, `Manifest(Old|New)` to make the code more readable and less error-prone. 

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by creativcoder — APPROVED** [review]

File: vm/state_migration/src/common/macros/mod.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod system;
+mod verifier;
+
+#[macro_export(local_inner_macros)]
```
Comment (LesnyRumcajs) [inline vm/state_migration/src/common/macros/mod.rs:7]: Could you briefly explain why we need the `local_inner_macros`?

File: vm/state_migration/src/nv18/eth_account.rs [hunk 2]
```diff
@@ -10,6 +10,8 @@ use forest_shim::{
 use forest_utils::db::BlockstoreExt;
 use fvm_ipld_blockstore::Blockstore;
 
+use super::*;
```
Comment (LesnyRumcajs) [inline vm/state_migration/src/nv18/eth_account.rs:13]: can we be more specific? I find looking for the objects imported with this cumbersome.

File: vm/state_migration/src/nv19/migration.rs [hunk 3]
```diff
@@ -5,19 +5,17 @@ use std::sync::Arc;
...
     clock::ChainEpoch,
-    machine::Manifest,
     state_tree::{StateTree, StateTreeVersion},
 };
 use forest_utils::db::BlockstoreExt;
 use fvm_ipld_blockstore::Blockstore;
 use fvm_ipld_encoding::CborStore;
 
-use super::{miner, power, system, verifier::Verifier};
+use super::{verifier::Verifier, *};
```
Comment (LesnyRumcajs) [inline vm/state_migration/src/nv19/migration.rs:18]: Same. Importing everything makes reading the code without an IDE at hand difficult.

**Review by LesnyRumcajs — COMMENTED** [review]
I like the idea. It'd be fantastic to add useful documentation on top of the macros. They tend to be less readable than _regular_ code so they deserve this special care.

File: vm/state_migration/src/common/macros/mod.rs (hunk 1 above)
Comment (hanabi1224) [reply vm/state_migration/src/common/macros/mod.rs:7]: I intended to use local macros inside local macros to further refactor some functions but it's not part of this PR, I can remove it from here and only add it later when needed

File: vm/state_migration/src/nv19/migration.rs (hunk 3 above)
Comment (hanabi1224) [reply vm/state_migration/src/nv19/migration.rs]: Fixed

File: vm/state_migration/src/nv18/migration.rs [hunk 4]
```diff
@@ -5,21 +5,17 @@ use std::sync::Arc;
...
     state_tree::{StateTree, StateTreeVersion},
 };
 use forest_utils::db::BlockstoreExt;
 use fvm_ipld_blockstore::Blockstore;
 use fvm_ipld_encoding::CborStore;
 
-use super::{
-    eam::create_eam_actor, eth_account::create_eth_account_actor, init, system, verifier::Verifier,
-};
+use super::{eam::create_eam_actor, eth_account::create_eth_account_actor, verifier::Verifier, *};
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply vm/state_migration/src/nv18/migration.rs]: Fixed

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**